mera/utils.py

```diff
@@ -0,0 +1,124 @@
+from typing import Optional
+import pandas as pd
+
+
+def mask_too_few_records(
+    residual_df: pd.DataFrame,
+    column_name1: Optional[str] = "event_id",
+    column_name2: Optional[str] = "stat_id",
+    min_num_records_per_column1: Optional[int] = 3,
+    min_num_records_per_column2: Optional[int] = 3,
+    verbose: Optional[bool] = False,
+) -> pd.DataFrame:
+    """
+    Mask records if there are too few records per event or per station to be useful.
+
+    Parameters
+    ----------
+    residual_df : pd.DataFrame
+        The residual DataFrame to mask.
+    column_name1 : str, default = 'event_id'
```

Comment:
You are correct, in theory this can be applied to any column, but given that this package is specifically for the standard(ish) use case for mixed-effects regression in our domain I think its best to just name these as `event_column` and `site_column`. Reduces risk of confusion.